extensions/liquibase-mongodb/deployment/src/main/java/io/quarkus/liquibase/mongodb/deployment/LiquibaseMongodbProcessor.java

```diff
@@ -154,82 +168,43 @@ void nativeImageConfiguration(
+        addService(services, reflective, liquibase.precondition.Precondition.class.getName(), true);
 
         // CommandStep implementations are needed (just like in non-mongodb variant)
-        addService(services, reflective, liquibase.command.CommandStep.class, false,
+        addService(services, reflective, liquibase.command.CommandStep.class.getName(), false,
                 "liquibase.command.core.StartH2CommandStep");
 
-        // liquibase XSD
-        resource.produce(new NativeImageResourceBuildItem(
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.7.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.8.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.9.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.10.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.11.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.12.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.13.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.14.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.15.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.16.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.17.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.18.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.19.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.20.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.21.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.22.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.23.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.24.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.25.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd",
-                "www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd",
-                "liquibase.build.properties"));
+        var dependencies = curateOutcome.getApplicationModel().getDependencies();
+
+        resource.produce(NativeImageResourceBuildItem.ofDependencyResources(
+                dependencies, LIQUIBASE_ARTIFACT, LIQUIBASE_PROPERTIES_FILTER));
+        resource.produce(NativeImageResourceBuildItem.ofDependencyResources(
+                dependencies, LIQUIBASE_ARTIFACT, LIQUIBASE_DB_CHANGELOG_FILTER));
+        resource.produce(NativeImageResourceBuildItem.ofDependencyResources(
+                dependencies, LIQUIBASE_MONGODB_ARTIFACT, LIQUIBASE_MONGODB_CHANGELOG_FILTER));
+        resource.produce(NativeImageResourceBuildItem.ofDependencyResources(
+                dependencies, LIQUIBASE_MONGODB_ARTIFACT, LIQUIBASE_MONGODB_PARSER_FILTER));
```

Comment:
It seems like it could have been somewhat simplified by combining all the filters applied to the same set of of artifacts into one. I mean `PathFilter.forIncludes(List.of(...))` could have been used instead, which would result in two calls to `resource.produce(NativeImageResourceBuildItem.ofDependencyResources(` here and one in the other place.

Then, personally, it doesn't seem like the new `NativeImageResourceBuildItem.ofDependencyResources()` offers a lot of benefit, since essentially, it wraps a one-liner `ArtifactResourceResolver.of(dependencies, artifactCoordinates).resourceList(resourceFilter)` and by that, from the caller's perspective, it takes the responsibility of filtering resources off of the `ArtifactResourceResolver`. I'll leave that decision to you though.

Reply:
I'll combine the resource filters. That was something I thought about during the modification and then promptly forgot about it. Thanks for the reminder,

I introduced the `NativeImageResourceBuildItem.ofDependencyResources` convenience method, to be on the same abstraction level as `ServiceProviderBuildItem allProvidersOfDependency`. It's where I would look first, to determine which capabilities are offered. In my opinion using the `ArtifactResourceResolver` is a bit more obscure.

I would keep the method, but since you have the better domain knowledge, I will remove it, if you think it's cleaner to use the `ArtifactResourceResolver` directly. What do you think?

Reply:
Sure, feel free to keep the method. Thanks.